src/Rules/Functions/ParamAttributesRule.php

```diff
@@ -27,11 +28,27 @@ public function getNodeType(): string
 
 	public function processNode(Node $node, Scope $scope): array
 	{
+		$targetName = 'parameter';
+		if (($node->flags & Class_::MODIFIER_PUBLIC) !== 0) {
```

Comment:
I don't get it - why only public?

Reply:
Also - if you rebase the branch, the build should succeed :) The problem was on master.

Reply:
I was trying to figure out how to know within this Rule that the node (`Node\Param`) is actually a promoted property (the scope didn't seem to have the right information). The PHP-Parser changelog led me to this solution: https://github.com/nikic/PHP-Parser/blob/master/CHANGELOG.md#added-7

If a function parameter has a public visibility, it is a promoted property since this is the only valid instance where a parameter could have a public visibility.

Reply:
What about private properties? It should work for them too. You can check whether `$node->flags` is non-zero, that will filter out promoted properties.

Reply:
Ah you're right... I totally forgot about other visibilities for some reason :)